kratos/geometries/nurbs_curve_on_surface_geometry.h

```diff
@@ -158,10 +158,10 @@ class NurbsCurveOnSurfaceGeometry : public Geometry<typename TCurveContainerPoin
     ) const override
     {
         // Compute the coordinates of the embedded curve in the parametric space of the surface
-        rResult = mpNurbsCurve->GlobalCoordinates(rResult, rLocalCoordinates);
+        CoordinatesArrayType result_local = mpNurbsCurve->GlobalCoordinates(rResult, rLocalCoordinates);
```

Comment:
I'm not sure if it's good to return to this value and pass the rResult.

I would define the variable before and then only pass it by reference.

Reply:
I agree, think this is legacy, not sure why it was done in the first place

Mybe @pooyan-dadvand has a comment

Reply:
I will write it on a local variable then

Reply:
After that change I we can merge it.

Reply:
Yes, it is a legacy design and my fault! :see_no_evil: 

Meanwhile, I don't understand why you need `result_local`?

Reply:
The idea is not to mix rResult with coming from mpNurbsCurve->GlobalCoordinates with the rResult coming as an input, cause they both need to be fed into mpNurbsSurface->GlobalCoordinates